api/cpp/include/slint-platform.h

```diff
@@ -630,6 +631,55 @@ class SoftwareRenderer : public AbstractRenderer
         return PhysicalRegion { r };
     }
 
+    /// Render the window scene, line by line. The provided Callback will be invoked for each line
+    /// that needs to rendered.
+    ///
+    /// The renderer uses a cache internally and will only render the part of the window
+    /// which are dirty.
+    ///
+    /// This function returns the physical region that was rendered considering the rotation.
+    ///
+    /// The callback is invoked with the line number as first parameter, and the start x and end x
+    /// coordinates of the line as second and third parameter. The implementation must provide a
+    /// line buffer (as std::span) and invoke the provided fourth function pointer (render_fn) with
+    /// it, to fill it with pixels. The last parameter of Callback is a private data pointer that
+    /// must be provided as first argument to render_fn.
+    /// After the line buffer is filled with pixels, your implementation is free to flush that line
+    /// to the screen for display.
+    template<std::invocable<std::size_t, std::size_t, std::size_t,
+                            void (*)(void *, std::span<Rgb565Pixel> &), void *>
```

Comment:
This is the public API, so it would be nice to avoid void*.
I'm not sure how to express that in a concept, but it should be anything callable with a span<Rgb565Pixel>

Reply:
I think the concept could be something like this:

template<typename F>
concept LineCallback = requires(F f, std::size_t line, std::size_t begin, std::size_t end) {
    { f(line, begin, end, [&y](const std::span<int>&) {}) } -> std::same_as<void>;
};

https://godbolt.org/z/5vobh5xWh